crates/astria-composer/src/executor/mod.rs

```diff
@@ -199,6 +203,16 @@ impl Executor {
     /// An error is returned if connecting to the sequencer fails.
     #[instrument(skip_all, fields(address = %self.address))]
     pub(super) async fn run_until_stopped(mut self) -> eyre::Result<()> {
+        let remote_chain_id = self
+            .get_sequencer_chain_id()
+            .await
+            .wrap_err("failed obtain sequencer chain_id")?;
+        ensure!(
+            self.sequencer_chain_id == remote_chain_id,
```

Comment:
You `remote_chain_id.as_str()` from https://docs.rs/tendermint/latest/tendermint/chain/id/struct.Id.html (important due to my suggestion to just return `tendermint::chain::Id` withot allocating).

Reply:
Resolved in 8523e04.